superset-frontend/src/components/IndeterminateCheckbox/IndeterminateCheckbox.test.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { render, screen } from 'spec/helpers/testing-library';
+import userEvent from '@testing-library/user-event';
+import IndeterminateCheckbox from '.';
+
+const mockedProps = {
+  checked: false,
+  id: 'checkbox-id',
+  indeterminate: false,
+  title: 'Checkbox title',
+  onChange: () => null,
+};
+
+jest.mock('../Icon', () => ({
```

Comment:
@geido I think you shouldn't mock the `Icon`. It's an integral part of this component, and you are making assertions about it. For example:

```
test('should render the checkbox-half icon', () => {
  const indeterminateProps = {
    ...mockedProps,
    indeterminate: true,
  };
  render(<IndeterminateCheckbox {...indeterminateProps} />);
  expect(screen.getByTestId('icon')).toBeInTheDocument();
  expect(screen.getByTestId('icon')).toHaveAttribute(
    'data-name',
    'checkbox-half',
  );
});
```
The attribute `data-name` only exists in your mock, but not in the real component. 

Take a look at this [PR](https://github.com/apache/superset/pull/13457). Here I added `role` and `aria-label` properties to the `Icon` component. You could base all your queries using these props. Unfortunately, it's not merged yet, but you can copy and paste it here.

If you prefer to mock the `Icon`, I would suggest only making assertions about properties that also exist on the real component 😉